crates/rattler_repodata_gateway/src/sparse/mod.rs

```diff
@@ -162,6 +173,10 @@ impl SparseRepoData {
                 )?;
                 records.append(&mut conda_records);
 
+                if strict_channel_priority && !records.is_empty() {
+                    found_in_channel = Some(repo_data.channel.base_url.clone());
+                }
```

Comment:
Just store a reference to forgo a clone:

```suggestion
                if strict_channel_priority && !records.is_empty() {
                    found_in_channel = Some(&repo_data.channel.base_url);
                }
```

Reply:
Nice!